app/src/main/java/com/github/lookupgroup27/lookup/ui/overview/Calendar.kt

```diff
@@ -1,5 +1,302 @@
+                  )
+
+                  if (hasEvents) {
+                    Spacer(modifier = Modifier.height(4.dp))
+                    Box(
+                        modifier =
+                            Modifier.size(8.dp)
+                                .background(MaterialTheme.colorScheme.primary, shape = CircleShape))
+                  }
+                }
+              }
+        }
+      }
+}
+
+@Composable
+fun EventList(calendarViewModel: CalendarViewModel, selectedDate: Date) {
+  val dateFormat = SimpleDateFormat("MMMM d, yyyy", Locale.getDefault())
+  val formattedDate = dateFormat.format(selectedDate)
+
+  val eventsForDay =
+      calendarViewModel.icalEvents.filter { event ->
+        val dtStart = event.getProperty<DtStart>(Property.DTSTART)?.date
+        dtStart != null && isSameDay(dtStart, selectedDate)
+      }
+
+  Column(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
+    Text(
+        text = "Events ($formattedDate):",
+        style = MaterialTheme.typography.titleMedium,
+        fontWeight = FontWeight.Bold,
+        modifier = Modifier.padding(bottom = 8.dp))
+
+    if (eventsForDay.isNotEmpty()) {
+      eventsForDay.forEach { event ->
+        EventItem(event = event, onClick = {})
+        Spacer(modifier = Modifier.height(8.dp))
+      }
+    } else {
+      Text(
+          text = "No events for today.",
+          style = MaterialTheme.typography.bodyMedium,
+          color = MaterialTheme.colorScheme.onSurfaceVariant)
+    }
+  }
+}
+
+@Composable
+fun EventItem(event: VEvent, onClick: () -> Unit) {
+  val eventTitle = event.getProperty<Summary>(Property.SUMMARY)?.value ?: "Unnamed Event"
+  val eventDate = event.getProperty<DtStart>(Property.DTSTART)?.date as? Date
+  val formattedDate =
+      eventDate?.let { SimpleDateFormat("MMMM d, yyyy", Locale.getDefault()).format(it) }
+          ?: "Unknown date"
+
+  Card(
+      modifier = Modifier.fillMaxWidth().clickable(onClick = onClick),
+      elevation = CardDefaults.cardElevation(4.dp),
+      colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface)) {
+        Column(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
+          Text(
+              text = eventTitle,
+              style = MaterialTheme.typography.bodyLarge,
+              fontWeight = FontWeight.SemiBold)
+          Spacer(modifier = Modifier.height(4.dp))
+
+          Text(
+              text = "Date: $formattedDate",
+              style = MaterialTheme.typography.bodySmall,
+              color = MaterialTheme.colorScheme.onSurfaceVariant)
+        }
+      }
+}
+
+fun isSameDay(date1: Date, date2: Date): Boolean {
+  val cal1 = Calendar.getInstance().apply { time = date1 }
+  val cal2 = Calendar.getInstance().apply { time = date2 }
+  return cal1.get(Calendar.YEAR) == cal2.get(Calendar.YEAR) &&
+      cal1.get(Calendar.DAY_OF_YEAR) == cal2.get(Calendar.DAY_OF_YEAR)
+}
+
+fun updateMonth(date: Date, months: Int): Date {
```

Comment:
Consider creating a `DateUtils` module to group all helper functions for dates, so that they can be reused in other files. That way you can also keep track of all already defined helper functions to avoid duplicates.